service-config/src/main/java/com/palantir/config/service/ServiceConfiguration.java

```diff
@@ -40,26 +40,23 @@
     public abstract Optional<SslConfiguration> security();
 
     /**
+     * Connect timeout for requests in ms.
+     */
+    public abstract Optional<Integer> connectTimeoutMs();
```

Comment:
Duration has a pretty funky serialization format; are you proposing we ask users to specify timeouts as something like "PT1M" (one minute), or that they specify milliseconds and the java class exposes Durations?